pandas/tests/io/conftest.py

```diff
@@ -37,11 +41,18 @@ def s3_resource(tips_file, jsonl_file):
     """
     pytest.importorskip('s3fs')
     boto3 = pytest.importorskip('boto3')
-    # GH-24092. See if boto.plugin skips the test or fails.
-    try:
-        pytest.importorskip("boto.plugin")
-    except AttributeError:
-        raise pytest.skip("moto/moto error")
+    botocore = pytest.importorskip('botocore')
+
+    if LooseVersion(botocore.__version__) < LooseVersion("1.11.0"):
+        # botocore leaks an uncatchable ResourceWarning before 1.11.0;
+        # see GH 23731 and https://github.com/boto/botocore/issues/1464
+        pytest.skip("botocore is leaking resources before 1.11.0")
```

Comment:
actually this skip is needed because `travis-27` runs an older boto (I just didn't see it in the `.yml` because its a transitive dependency of `s3fs`).